app/src/main/java/it/niedermann/owncloud/notes/branding/NotesViewThemeUtils.java

```diff
@@ -0,0 +1,157 @@
+package it.niedermann.owncloud.notes.branding;
+
+import static com.nextcloud.android.common.ui.util.ColorStateListUtilsKt.buildColorStateList;
+import static com.nextcloud.android.common.ui.util.PlatformThemeUtil.isDarkMode;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.LayerDrawable;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.ColorInt;
+import androidx.annotation.IdRes;
+import androidx.annotation.NonNull;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.drawable.DrawableCompat;
+
+import com.google.android.material.appbar.AppBarLayout;
+import com.google.android.material.appbar.MaterialToolbar;
+import com.nextcloud.android.common.ui.theme.MaterialSchemes;
+import com.nextcloud.android.common.ui.theme.ViewThemeUtilsBase;
+import com.nextcloud.android.common.ui.theme.utils.MaterialViewThemeUtils;
+
+import it.niedermann.android.util.ColorUtil;
+import it.niedermann.owncloud.notes.R;
+import it.niedermann.owncloud.notes.main.navigation.NavigationItem;
+import it.niedermann.owncloud.notes.shared.util.NotesColorUtil;
+import kotlin.Pair;
+import scheme.Scheme;
+
+public class NotesViewThemeUtils extends ViewThemeUtilsBase {
+
+    private static final String TAG = NotesViewThemeUtils.class.getSimpleName();
+    private final MaterialSchemes schemes;
+
+    public NotesViewThemeUtils(@NonNull MaterialSchemes schemes) {
+        super(schemes);
+        this.schemes = schemes;
+    }
+
+    @ColorInt
+    public int getOnPrimaryContainer(@NonNull Context context) {
+        return withScheme(context, Scheme::getOnPrimaryContainer);
+    }
```

Comment:
One of the design goals of the ui theming lib is to prevent client apps from having to deal with the schemes directly, instead calling the available theming utils for each specific view type and using `ColorRole`s if they need to choose a specific color. I'll try to figure out what is needed to avoid this method and add it to common:ui.

Reply:
That being said, I'm okay with merging without changing this in the interest of moving forward, but we should add a `@Deprecated` to this method explaining the above.

Reply:
After https://github.com/nextcloud/android-common/pull/77 is merged, I believe all of the usages of this method can be removed in favor of using `ColorRole.ON_PRIMARY_CONTAINER` where appropriate.

Reply:
I replaced it where possible, but one point is still missing: We are creating `ForegroundColorSpan`s for highlighting the search term in results. I haven't been able to replace this quickly, so I kept the method, added a `@Deprecated` annotation as you suggested.

Reply:
Is this enough? https://github.com/nextcloud/android-common/pull/78/files 

If not just leave it as dupe